src/dbus/mpris.rs

```diff
@@ -145,11 +146,44 @@ impl SpotMprisPlayer {
     }
 
     pub fn seek(&self, Offset: i64) -> Result<()> {
-        Ok(())
+        if !self.can_seek() {
+            return Ok(());
+        }
+
+        let pos: u32 = (self.state.position() / 1000)
+            .try_into()
+            .map_err(|_| zbus::fdo::Error::Failed("Could not parse position".to_string()))?;
+
+        let offset: u32 = (Offset / 1000)
+            .try_into()
+            .map_err(|_| zbus::fdo::Error::Failed("Could not parse position".to_string()))?;
+
+        self.sender
+            .unbounded_send(PlaybackAction::Seek(pos + offset).into())
+            .map_err(|_| zbus::fdo::Error::Failed("Could not send action".to_string()))
     }
 
-    fn set_position(&self, TrackId: String, Position: i64) -> Result<()> {
-        Ok(())
+    pub fn set_position(&self, TrackId: ObjectPath, Position: i64) -> Result<()> {
+        if !self.can_seek() {
+            return Ok(());
+        }
+
+        if TrackId.to_string() != self.metadata().id {
+            return Ok(());
+        }
+
+        // FIXME: Implicit position < 0 check by casting to u32
+        let pos: u32 = (Position / 1000)
+            .try_into()
+            .map_err(|_| zbus::fdo::Error::Failed("Could not parse position".to_string()))?;
+
+        if pos as u64 > self.metadata().length / 1000 {
```

Comment:
Maybe compare Position and metadata().length earlier on? To avoid this division (and maybe a cast too?)

Reply:
Unfortunately I think a cast is necessary here, but moving can avoid a division.